### PR TITLE
Refactor/managing dependencies

### PR DESCRIPTION
**Pull request by Nicoss54**
This PR let to have one source of trust of dependencies, like a mono repo

This PR fix also the use of jasmine spyOn instead spy on of jest in the promise steps

### CODE REVIEW

File: steps/promises/test.spec.js [hunk 1]
```diff
@@ -1,39 +1,41 @@
-const {exercise1Fn, exercise2Fn, exercise3Fn} = require('./exo.js');
+const { exercise1Fn, exercise2Fn, exercise3Fn } = require('./exo.js');
 
 describe('Test exercise for promises', function () {
-
-  let instanceToTest;
+  let spyConsoleLog = undefined;
```
Comment (kuroidoruido) [inline steps/promises/test.spec.js:4]: C'est quoi l'apport de passer sur la syntaxe Jest ?

File: package.json [hunk 2]
```diff
@@ -45,13 +50,19 @@
...
     "typescript": "^4.1.3"
   },
   "dependencies": {
     "axios": "^0.18.0",
     "jest": "^26.6.3",
     "ramda": "^0.26.1",
     "rollup": "^4.18.0",
-    "ts-jest": "^26.4.4"
+    "ts-jest": "^26.4.4",
+    "sfeir-school-theme": "^3.1.0"
```
Comment (kuroidoruido) [inline package.json]: ça apporte quoi de tout fusionner ? c'était bien de tout séparer je trouve et au pire avoir un package.json à la racine qui sert de workspace

File: steps/promises/test.spec.js (hunk 1 above)
Comment (Nicoss54) [reply steps/promises/test.spec.js:4]: Nothing in terms of performance or in the same area, it's just to be coherent with documentation and stackoverflow doc

File: package.json (hunk 2 above)
Comment (Nicoss54) [reply package.json]: great idea, let me refactor buddy, @kuroidoruido you rock !

**Comment by Nicoss54** [comment]
@jefBinomed @kuroidoruido i move the architecture of this project to match a npm workspace, sorry for all the delete files

File: package.json (hunk 2 above)
Comment (kuroidoruido) [reply package.json]: IMHO: no need to move everything into an apps folder. This move only create noise and does make the repo less clear for students... You can do the same thing without moving steps and docs directories into an app folder and keep this repo uniform with the template and other schools

File: docs/web_modules/sfeir-school-theme/README.md [hunk 3]
```diff
@@ -1,1046 +0,0 @@
-[![npm version](https://badge.fury.io/js/sfeir-school-theme.svg)](https://badge.fury.io/js/sfeir-school-theme)
```
Comment (kuroidoruido) [inline docs/web_modules/sfeir-school-theme/README.md:1]: You should not drop the web_modules directory, which is needed for the github pages version

**Pull request merged**